fastparquet/test/test_writer.py

```diff
@@ -272,6 +283,23 @@ def test_groups_roundtrip(tempdir):
         assert row.b in list(df[(df.a==row.a)&(df.c==row.c)].b)
 
 
+def test_empty_groupby(tempdir):
+    df = pd.DataFrame({'a': np.random.choice(['a', 'b', None], size=1000),
+                       'b': np.random.randint(0, 64000, size=1000),
+                       'c': np.random.choice([True, False], size=1000)})
+    df.loc[499:, 'c'] = True  # no False in second half
+    writer.write(tempdir, df, partition_on=['a', 'c'], file_scheme='hive',
+                 partitions=[0, 500])
+    r = ParquetFile(tempdir)
+    assert r.count == sum(~df.a.isnull())
+    assert len(r.row_groups) == 6
+    out = r.to_pandas()
```

Comment:
Well, ideally we don't create 1000 files for the user.  Looking briefly at the API I think that if I have a dataset that is well organized that I should add the organization columns to this input.  What happens if I do that?  Do I get 1000 files?  Do I get an informative error?  Do I get a well organized 10 files?  

If the current behavior is that I get 1000 errors then I think that that should be changed.